starport/services/scaffolder/module.go

```diff
@@ -213,36 +267,43 @@ func moduleExists(appPath string, moduleName string) (bool, error) {
 		return false, nil
 	}
 
-	return true, err
+	return err == nil, err
 }
 
-func checkModuleName(moduleName string) error {
+// checkModuleName checks if the name can be used as a module name
+func checkModuleName(appPath, moduleName string) error {
 	// go keyword
 	if token.Lookup(moduleName).IsKeyword() {
 		return fmt.Errorf("%s is a Go keyword", moduleName)
 	}
 
-	// name of default registered module
-	switch moduleName {
-	case
-		"auth",
-		"genutil",
-		"bank",
-		"capability",
-		"staking",
-		"mint",
-		"distr",
-		"gov",
-		"params",
-		"crisis",
-		"slashing",
-		"ibc",
-		"upgrade",
-		"evidence",
-		"transfer",
-		"vesting":
-		return fmt.Errorf("%s is a default module", moduleName)
+	// check if the name is a reserved name
+	if _, ok := reservedNames[moduleName]; ok {
+		return fmt.Errorf("%s is a reserved name and can't be used as a module name", moduleName)
 	}
+
+	// check if the name can imply potential store key collision
+	for _, defaultStoreKey := range defaultStoreKeys {
+		if strings.HasPrefix(moduleName, defaultStoreKey) {
```

Comment:
can we reuse this logic at least to avoid dup, including the err dup of err msg.